Consistently use gfm rather than markdown_github

peterdesmet opened the pull request:
## Description

Some `index.Rmd` files used the deprecated `variant: markdown_github`, which results in a warning when knitting. This PR updates all `index.Rmd` files to consistently use the recommended `gfm` and 2-space indentation in yaml frontmatter:

```
output: 
  md_document:
    preserve_yaml: true
    variant: gfm
```

For some tutorials, this updates the `index_files/figure-markdown_github` to `index_files/gfm`. The PR has little to no effect on the website.

florisvdh commented on content/tutorials/r_citations_markdown/index.Rmd:
````diff
@@ -118,7 +118,7 @@ To specify the style file for the Research Institute for Nature and Forest it is
     ```
 
-### Other YAML citation arguments {#other-yaml-citation-arguments}
+### Other YAML citation arguments {\#other-yaml-citation-arguments}
````
I suggest to revert this change, it doesn't follow the recommended markdown syntax.

florisvdh commented on content/tutorials/r_citations_markdown/index.Rmd:
```diff
@@ -190,20 +190,19 @@ The defaults are OK.
 
-# Citing documents {#citing-documents}
-
+# Citing documents {\#citing-documents}
```
I suggest to revert this change, it doesn't follow the recommended markdown syntax.

florisvdh commented on content/tutorials/r_citations_markdown/index.Rmd:
```diff
@@ -190,20 +190,19 @@ The defaults are OK.
-| delineate multiple authors with colon: `[@adams1975; @aberdeen1958]` | delineate multiple authors with colon: [@adams1975; @aberdeen1958] |
+| syntax                                                               | result (depending on chosen csl style)                                 |
+|----------------------------------------------------------------------|------------------------------------------------------------------------|
+| `@adams1975` concludes that ...                                      | \@adams1975 concludes that ...                                         |
+| `@adams1975[p.33]` concludes that ...                                | \@adams1975 \[p.33\] concludes that ...                                |
+| ... end of sentence `[@adams1975]`.                                  | ... end of sentence \[\@adams1975\].                                   |
+| ... end of sentence `[see @adams1975,p.33]`.                         | ... end of sentence \[see \@adams1975,p.33\].                          |
+| delineate multiple authors with colon: `[@adams1975; @aberdeen1958]` | delineate multiple authors with colon: \[\@adams1975; \@aberdeen1958\] |
```
Please revert this change, it doesn't follow the recommended markdown syntax.

florisvdh commented on content/tutorials/r_citations_markdown/index.Rmd:
````diff
@@ -238,7 +237,7 @@ For instance:
 ```
 
-... allows us to cite @R-base, @R-bookdown and @R-rmarkdown if we add `packages.bib` to the bibliography YAML field.
+... allows us to cite \@R-base, \@R-bookdown and \@R-rmarkdown if we add `packages.bib` to the bibliography YAML field.
````
Please revert this change, it doesn't follow the recommended markdown syntax.

florisvdh reviewed (changes requested):
All seems fine, except in one Rmd file where some characters were escaped for an unknown reason. I suggest to revert that since it is not consistent with Pandoc's markdown.

florisvdh reviewed (commented):
Further, see https://github.com/inbo/tutorials/issues/200#issuecomment-988006466. I advise to use `gfm+footnotes` instead of `gfm`.

Thanks for the efforts you have been doing here @peterdesmet :+1:

florisvdh commented:
> I advise to use `gfm+footnotes` instead of `gfm`.

In order not to have to re-knit tutorials, you could just update the Rmd files here, so that this is taken care of next time they are knitted. But probably that is your approach already.

Caveat: some open pull requests contain new tutorials, so the changes may have to be applied there as well.

florisvdh commented:
> Caveat: some open pull requests contain new tutorials, so the changes may have to be applied there as well.

In order to limit this problem, can you have a further look at remaining tasks in this PR @peterdesmet, so that we get this in `master`? Thanks!

peterdesmet commented:
@florisvdh the backslashes are removed. Do you want me to use `gfm+footnotes` everywhere or in specific places?

florisvdh commented:
> @florisvdh the backslashes are removed. Do you want me to use `gfm+footnotes` everywhere or in specific places?

I suggest to do it everywhere, since often a new tutorial will be based on another one and we better do it consistently. If you wish, I could try to do this at once, with `sed`. Further, such update is best applied to existing PRs with new tutorials, not only `master`. Maybe the 'create tutorial' tutorial also needs checking.

If you'd find a way to let Hugo make the citation links that appear useful - i.e. the links which appear after adding `link-citations: true` - then that would be superb! I didn't find the solution in https://github.com/inbo/tutorials/issues/200#issuecomment-988006466.

peterdesmet commented:
@florisvdh I have
- Removed `output` frontmatter from `index.md` files that are not generated from `index.Rmd`
- Updated `index.Rmd` frontmatter to use `gfm+footnotes` and knitted where relevant. See e.g. footnotes at bottom, where they weren't there before at https://inbo.github.io/tutorials/tutorials/spatial_crs_coding/
- Updated `create a tutorial` to indicate the desired frontmatter

peterdesmet commented:
And finally, I have updated the frontmatter in the open pull requests, so we don't reintroduce old frontmatter.

florisvdh reviewed (approved):
Perfect, well done :+1: . Thank you!
